src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -972,6 +976,40 @@
               :subquery? false
               :source-table "producers"}))
 
+(def packages-query
+  "Basic packages query"
+  (map->Query {:projections {"certname" {:type :string
+                                         :queryable? true
+                                         :field :certnames.certname}
+                             "package_name" {:type :string
+                                             :queryable? true
+                                             :field :pi.name}
+                             "version" {:type :string
+                                        :queryable? true
+                                        :field :pi.version}
+                             "provider" {:type :string
+                                         :queryable? true
+                                         :field :pi.provider}}
+
+               :selection {:from [[:package_inventory :pi]]
+                           :left-join [:certnames
+                                       [:= :pi.certname_id :certnames.id]]}
```

Comment:
the CTE we just introduced will flip the semantics of these left joins (assuming we have others) when the certnames table is empty -- before, that would have excluded all results; now, it will include all results. Not sure what to do about that, or whether we care.